File: app/src/main/java/com/futuretech/nfmovies/API/ISite.java
package com.futuretech.nfmovies.API;

import com.futuretech.nfmovies.Entity.Category;
import com.futuretech.nfmovies.Entity.Episode;
import com.futuretech.nfmovies.Entity.Movie;

import java.util.List;

public interface ISite {

    List<Category> getCategories() throws Exception;
    Site getSite();
    Movie getMovieDetail(Movie movie) throws Exception;
    String getPlayURL(Episode e) throws Exception;
    Category search(String str) throws Exception;
    String getName();
    String getHost();
}

File: app/src/main/java/com/futuretech/nfmovies/API/ISite.kt
package com.futuretech.nfmovies.API

import com.futuretech.nfmovies.Entity.Category
import com.futuretech.nfmovies.Entity.Episode
import com.futuretech.nfmovies.Entity.Movie

interface ISite {

    val categories: List<Category>
    val site: Site
    val name: String
    val host: String
    @Throws(Exception::class)
    fun getMovieDetail(movie: Movie): Movie

    @Throws(Exception::class)
    fun getPlayURL(e: Episode): String

    @Throws(Exception::class)
    fun search(str: String): Category
}
